streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -286,17 +286,17 @@ public StreamThread(TopologyBuilder builder,
                         Time time,
                         StreamsMetadataState streamsMetadataState,
                         final long cacheSizeBytes) {
-        super("StreamThread-" + STREAM_THREAD_ID_SEQUENCE.getAndIncrement());
+        super(clientId + "-StreamThread-" + STREAM_THREAD_ID_SEQUENCE.getAndIncrement());
         this.applicationId = applicationId;
-        String threadName = getName();
+        final String threadName = getName();
```

Comment:
I guess, we can get rid of local var `threadName` and just use `threadClientId`.